packages/stream_video/lib/src/call/permissions/permissions_manager.dart

```diff
@@ -313,6 +313,42 @@ class PermissionsManager {
     );
   }
 
+  Future<Result<None>> pinForEveryone({
+    required String userId,
+    required String sessionId,
+  }) async {
+    if (!hasPermission(CallPermission.pinForEveryone)) {
+      _logger.w(() => '[pinForEveryone] rejected (no permission)');
+      return Result.error('Cannot pin session for everyone (no permission)');
+    }
+
+    _logger.d(() => '[pinForEveryone] pinning $userId for everyone');
+
+    return coordinatorClient.videoPin(
+      callCid: callCid,
+      sessionId: sessionId,
+      userId: userId,
+    );
```

Comment:
Not for this PR, as it already happens in this file, but it feels weird to me that a permission manager actually executes the request. Either we could rename this permission manager, or change how this works. If I call `pinForEveryone` on a permission manager I would not expect that thing to happen directly.

Reply:
Yep agreed. Tomasso also suggested that we shouldn't check permissions at all and leave it to the backend. Something to consider.